### DISCUSSION TITLE
Redirect URL in Github Authorization Not Working

### DISCUSSION THREAD

**Discussion by QuentinRoggy** [supabase/supabase#17374] [Contribute to Supabase]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

I'm actually working on the tutorial provided by eggHead.io on twitter clone with supabase & nextJs. 
I setup a login button like and it's look work fine because i'm redirected to github oath. But after, redirect does not work. I have a 500 error and no more explaination.

## To Reproduce
This is my button code : 
`
"use client";

import { createClientComponentClient } from "@supabase/auth-helpers-nextjs";

export default function AuthButton() {
  const supabase = createClientComponentClient();

  const handleSignOut = async () => {
    await supabase.auth.signOut();
  };

  const handleSignIn = async () => {
    await supabase.auth.signInWithOAuth({
      provider: "github",
      options: {
        redirectTo: "http://localhost:3000/auth/callback",
      },
    });
  };

  return (
    <>
      <button onClick={handleSignIn}>Login</button>
      <button onClick={handleSignOut}>Logout</button>
    </>
  );
}
`

and here the route code : 
`
import { createRouteHandlerClient } from "@supabase/auth-helpers-nextjs";
import { cookies } from "next/headers";

import { NextResponse, type NextRequest } from "next/server";

export async function GET(request: NextRequest) {
  const requestUrl = new URL(request.url);
  const code = requestUrl.searchParams.get("code");

  if (code) {
    const supabase = createRouteHandlerClient({ cookies });
    await supabase.auth.exchangeCodeForSession(code);
  }

  return NextResponse.redirect(requestUrl.origin);
}
`

## Expected behavior

Redirect back to the http://localhost:3000/ page after the GitHub Authorization.

## System information

- OS: macOS
- Browser chrome
- Version of supabase-js: 2.33.2
- Version of Node.js: 18.2.0

**Comment by saltcod** [comment]
cc @dijonmusters if you've got any immediate ideas
